### DISCUSSION TITLE
How to fall back to None value on ValidationError

### DISCUSSION THREAD

**Discussion by johtso** [pydantic/pydantic#2757] [Question] [answered]
I have a field that I am constraining using a `Optional[Literal["A", "B", "C"]]` type.

The incoming data may contain other values that I'm not interested in.

In that case I just want the value to be `None` rather than throwing a `ValidationError`.

What's the right way to do this? Is there a way to specify that the field should perform "lenient" validation? This doesn't seem like all that uncommon a thing to want to do.

I obviously thought about using a `pre=True` validator.. but it's not clear how to do that without it being really clunky. I basically need to manually perform `pydantic`s default validation behaviour for that data, catch the exception, and return `None` or the value. Is this described at all in the docs?

**Comment by nymous** [comment]
Hello!
That looks a lot like the question in #2741, you could reuse the solution there

**Reply by johtso** [reply]
Thanks for that @nymous ! 

Looks close to achieving what I want, but I'm not trying to manually parse a Model, I'm just trying to manually parse/validate a single field.

I think that would require doing something different?

**Comment by samuelcolvin** [comment, marked as answer]
Sadly, there's no easy way to do this until we get #1984 or similar.

**Reply by johtso** [reply]
Thanks for clarifying that! It's a tricky onion to wrestle :)
